src/test/java/unit/BqDatasetNumTables.java

```diff
@@ -153,5 +176,167 @@ void numTablesForReferencedWithNoAccess() throws IOException {
+        "workspace", "add-user", "--email=" + shareeUser.email, "--role=READER");
+
+    // `terra resource add-ref bq-dataset --name=$name --project-id=$projectId
+    // --dataset-id=$datasetId`
+    String name = "updateTableReferenceWithNoAccess";
+    TestCommand.runCommandExpectSuccess(
+        "resource",
+        "add-ref",
+        "bq-table",
+        "--name=" + name,
+        "--project-id=" + externalDataset.getProjectId(),
+        "--dataset-id=" + externalDataset.getDatasetId(),
+        "--table-id=" + privateExternalTable);
+
+    shareeUser.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+
+    String newName = "updateTableReferenceWithNoAccess_NEW";
+    TestCommand.runCommandExpectExitCode(
+        2, "resource", "update", "bq-table", "--name=" + name, "--new-name=" + newName);
+
+    // Clean up.
+    workspaceCreator.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+    TestCommand.runCommandExpectSuccess("resource", "delete", "--name=" + name, "--quiet");
+    TestCommand.runCommandExpectSuccess("workspace", "delete", "--quiet");
+  }
+
+  @Test
+  @DisplayName(
+      "Attempt to update table reference when the user only have access to sharedExternalTable.")
+  void updateTableReferenceWithPartialAccess() throws IOException {
+    workspaceCreator.login();
+
+    UFWorkspace createWorkspace =
+        TestCommand.runAndParseCommandExpectSuccess(UFWorkspace.class, "workspace", "create");
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+    // `terra workspace add-user --email=$email --role=READER`
+    TestCommand.runCommandExpectSuccess(
+        "workspace", "add-user", "--email=" + shareeUser.email, "--role=WRITER");
+
+    // `terra resource add-ref bq-dataset --name=$name --project-id=$projectId
+    // --dataset-id=$datasetId`
+    String name = "updateTableReferenceWithPartialAccess";
+    TestCommand.runCommandExpectSuccess(
+        "resource",
+        "add-ref",
+        "bq-table",
+        "--name=" + name,
+        "--project-id=" + externalDataset.getProjectId(),
+        "--dataset-id=" + externalDataset.getDatasetId(),
+        "--table-id=" + sharedExternalTable);
+
+    shareeUser.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+
+    String newName = "updateTableReferenceWithPartialAccess_NEW";
+    UFBqTable updateTable =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFBqTable.class,
+            "resource",
+            "update",
+            "bq-table",
+            "--name=" + name,
+            "--new-name=" + newName);
+    assertEquals(newName, updateTable.name);
+
+    TestCommand.runCommandExpectExitCode(
+        1,
```

Comment:
why is this a different exit code that the other failures? perhaps you can assert against the exception message to make that clearer